pkg/build/ci.go

```diff
@@ -0,0 +1,177 @@
+	if buildExistsErr != nil {
+		return errors.Wrapf(buildExistsErr, "checking if build exists")
+	}
+
+	if buildExists {
+		return nil
+	}
+
+	logrus.Infof("Proceeding with a new build...")
+
+	// TODO: Should this be configurable?
+	envErr := os.Setenv("KUBE_RELEASE_RUN_TESTS", "n")
+	if envErr != nil {
+		return errors.Wrapf(envErr, "setting 'KUBE_RELEASE_RUN_TESTS' to 'n'")
+	}
+
+	// Configure docker client for gcr.io authentication to allow communication
+	// with non-public registries.
+	if bi.opts.ConfigureDocker {
+		if configureErr := auth.ConfigureDocker(); configureErr != nil {
+			return errors.Wrapf(configureErr, "configuring docker auth")
+		}
+	}
+
+	if cleanErr := command.New(
+		"make",
+		"clean",
+	).RunSuccess(); cleanErr != nil {
+		return errors.Wrapf(cleanErr, "running make clean")
+	}
+
+	// Create a Kubernetes build
+	releaseType := "release"
+	if bi.opts.Fast {
+		releaseType = "quick-release"
+	}
+
+	if buildErr := command.New(
+		"make",
+		releaseType,
+	).RunSuccess(); buildErr != nil {
+		return errors.Wrapf(buildErr, fmt.Sprintf("running make %s", releaseType))
+	}
+
+	// Pushing the build
+	return bi.Push()
+}
+
+// checkBuildExists check if the target build exists in the specified GCS
+// bucket. This allows us to speed up build jobs by not duplicating builds.
+func (bi *Instance) checkBuildExists() (bool, error) {
+	version, getVersionErr := release.GetWorkspaceVersion()
+	if getVersionErr != nil {
+		return false, errors.Wrap(getVersionErr, "getting workspace version")
+	}
+
+	bi.opts.Version = version
+	if bi.opts.Version == "" {
+		logrus.Infof("Failed to get a build version from the workspace")
+		return false, nil
+	}
+
+	bucket := bi.opts.Bucket
+	if bi.opts.Bucket == "" {
+		bucket = "kubernetes-release-dev"
+	}
+
+	mode := "ci"
+	if bi.opts.CI {
+		mode = "ci"
+	}
+
+	if bi.opts.Fast {
+		mode += "/fast"
+	}
+
+	if bi.opts.GCSSuffix != "" {
+		mode += bi.opts.GCSSuffix
+	}
+
+	bi.opts.Bucket = bucket
+
+	gcsBuildRoot := filepath.Join(bucket, mode, version)
+	kubernetesTar := filepath.Join(gcsBuildRoot, release.KubernetesTar)
+	binPath := filepath.Join(gcsBuildRoot, "bin")
+
+	gcsBuildRoot = gcs.NormalizeGCSPath(gcsBuildRoot)
+	kubernetesTar = gcs.NormalizeGCSPath(kubernetesTar)
+	binPath = gcs.NormalizeGCSPath(binPath)
+
+	gcsBuildPaths := []string{
+		gcsBuildRoot,
+		kubernetesTar,
+		binPath,
+	}
+
+	// TODO: Do we need to handle the errors more effectively?
+	existErrors := []error{}
+	for i, path := range gcsBuildPaths {
+		logrus.Infof("Checking if GCS build path (%s) exists", path)
+		exists, existErr := gcs.PathExists(path)
+		if existErr != nil || !exists {
+			existErrors = append(existErrors, existErr)
+		}
+
+		if i == len(gcsBuildPaths)-1 && len(existErrors) == 0 {
+			logrus.Infof("Build already exists. Exiting...")
+			return true, nil
+		}
+	}
+
+	logrus.Infof("The following error(s) occurred while looking for a build: %v", existErrors)
```

Comment:
What is the reason that we do not return the errors (maybe stacked/concatenated) `if len(existErrors) > 0`?